var/spack/repos/jcsda-emc-bundles/packages/ufs-weather-model-env/package.py

```diff
@@ -22,22 +22,39 @@ class UfsWeatherModelEnv(BundlePackage):
         description="Build a debug version of certain dependencies (ESMF, MAPL)",
     )
     variant("python", default=True, description="Build Python dependencies")
+    variant("shared", default=True, description="Build dynamic libraries when possible")
 
-    depends_on("base-env", type="run")
     depends_on("ufs-pyenv", type="run", when="+python")
 
-    depends_on("fms@2022.01", type="run")
-    depends_on("bacio", type="run")
-    depends_on("crtm@2.4.0", type="run")
-    depends_on("g2", type="run")
-    depends_on("g2tmpl", type="run")
-    depends_on("ip", type="run")
-    depends_on("sp", type="run")
-    depends_on("w3nco", type="run")
-
-    depends_on("esmf~debug", type="run", when="~debug")
-    depends_on("esmf+debug", type="run", when="+debug")
-    depends_on("mapl~debug", type="run", when="~debug")
-    depends_on("mapl+debug", type="run", when="+debug")
-
-    # There is no need for install() since there is no code.
+    depends_on("bacio")
+    depends_on("fms")
+    depends_on("crtm")
+    depends_on("g2")
+    depends_on("g2tmpl")
+    depends_on("ip")
+    depends_on("sp")
+    depends_on("w3emc")
+    depends_on("parallelio+fortran~pnetcdf~shared")
```

Comment:
Wondering if this can be simplified ... create a variant `shared` for the `ufs-weather-model-env` which can be `False` by default. Then:
```
if "~shared" in self.spec:
   variant = "~shared"
else:
   variant = "+shared"
depends_on("parallelio+fortran~pnetcdf{}".format(variant))
```
I haven't actually tried this, but would be interesting to see if it works.